policy/bundle.go

```diff
@@ -596,7 +600,7 @@ func (c *bundleCache) precompileQuery(query *explorer.Mquery, policy *Policy) {
 	if policy == nil {
 		c.lookupQuery[query.Mrn] = query
 	} else if existing, ok := c.lookupQuery[query.Mrn]; ok {
-		query.AddBase(existing)
+		query = query.Merge(existing)
```

Comment:
AddBase modifies the query in the policy group - we don't want that. Merge doesn't modify it, but we need the merged query a bit later for the compile step, so we don't want to drop it.